sphinx_search/extension.py

```diff
@@ -24,21 +30,39 @@ def copy_asset_files(app, exception):
 
 
 def inject_static_files(app):
-    """Inject correct CSS and JS files based on the value of ``RTD_SPHINX_SEARCH_FILE_TYPE``."""
-    file_type = app.config.rtd_sphinx_search_file_type
-    expected_file_type = ASSETS_FILES.keys()
+    """
+    Inject correct CSS and JS files based on the value of ``RTD_SPHINX_SEARCH_FILE_TYPE``.
+
+    This only injects file if the docs are build on Read the Docs.
+    """
+
+    # This is done to properly test the extension.
+    # If the tests are being executed, add `html`
+    # as one of the `ONLINE_BUILDERS`, so that
+    # proper CSS and JS files gets injected and can be tested.
+    if os.environ.get('TESTING_ENVIRONMENT') == 'True':
```

Comment:
I think is better is you test the extension installing the rtd sphinx extension

Reply:
I don't think that will help. We again ahve to modify something to mock the build environment in RTD.

Reply:
We use this in the other extension https://github.com/readthedocs/readthedocs-sphinx-ext/blob/master/tests/util.py

Reply:
Yea, I don't like having this in the code. There are other ways we can test this.

Reply:
@stsewd 
Installing readthedocs-sphinx-ext was a nice idea.
It removed unnecessary code.